sensor_test.py

```diff
@@ -0,0 +1,172 @@
+    wait(func=get_sensor_reading,
+         condition=lambda x: isinstance(x, dict),
+         tries=5,
+         timeout=new_sensor_reading_interval)
+
+    print("Step 6", "Get sensor reading")
+    sensor_reading_step6 = get_sensor_reading()
+
+    print("Step 7", "Validate that reading from Step 4 doesn't equal reading from Step 6")
+    assert sensor_reading_step6 != sensor_reading_step4
+
+
+# Максимальна версія прошивки сенсора -- 15
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+    max_firmware_version = 15
+
+    print("Step 1" "Get original sensor firmware version")
+    original_sensor_fw_ver = get_sensor_info().get("firmware_version")
+
+    print("Step 2" "Request firmware update")
+    sensor_fw_upgrade_request = update_sensor_firmware()
+    assert sensor_fw_upgrade_request == "updating"
+    wait(func=get_sensor_info, condition=lambda x: isinstance(x, dict), tries=15, timeout=3)
+
+    print("Step 3" "Get current sensor firmware version")
+    current_sensor_fw_ver = get_sensor_info().get("firmware_version")
+
+    print("Step 4" "Validate that current firmware version is +1 to original firmware version")
+    assert current_sensor_fw_ver == (original_sensor_fw_ver + 1)
+
+    print("Step 5" "Repeat steps 1-4 until sensor is at max_firmware_version - 1")
+    while current_sensor_fw_ver < max_firmware_version - 1:
+        sensor_fw_upgrade_request = update_sensor_firmware()
+        assert sensor_fw_upgrade_request == "updating"
+        wait(func=get_sensor_info, condition=lambda x: isinstance(x, dict), tries=15, timeout=3)
+
+        current_sensor_fw_ver = get_sensor_info().get("firmware_version")
+
+    print("Step 6" "Update sensor to max firmware version")
+    sensor_fw_upgrade_request = update_sensor_firmware()
+    assert sensor_fw_upgrade_request == "updating"
+    wait(func=get_sensor_info, condition=lambda x: isinstance(x, dict), tries=15, timeout=3)
+
+    print("Step 7" "Validate that sensor is at max firmware version")
+    assert get_sensor_info().get("firmware_version") == max_firmware_version
+
+    print("Step 8" "Request another firmware update")
+    sensor_fw_upgrade_request = update_sensor_firmware()
+
+    print("Step 9" "Validate that sensor doesn't update and responds appropriately")
+    assert sensor_fw_upgrade_request != "updating"
```

Comment:
This check isn't needed because you check for `already at latest firmware version` below.

Reply:
assert sensor_fw_upgrade_request != "updating"  - unnecessary check was removed